### DISCUSSION TITLE
Render object with plain material are drawn multiple times (and appears brighter)

### DISCUSSION THREAD

**Discussion by dlyr** [STORM-IRIT/Radium-Engine#751] [Design discussion]
Plain material is suppose to draw without lighting. But it has some issue with ForwardRenderer.

It's mostly used for Ui or Debug, which seems to work fine (since there is no lighting in these rendering).
But if it's a geometry component, the ro is added to the light loop, even if it has plain material.

On simple solution is make plain material only affect ambient/zprepass, but it's not used in ui/debug rendering.
Another solution is to refactor the way ForwardRenderer choose how to draw things.
Or simply wait for the "new renderer" that can fix this (hopefully?)

**Comment by MathiasPaulin** [comment]
I think that we shouldn't design any Material behavior in a way depending on a specific renderer (neither `Forward` nor `new Renderer` nor ... ) or on a RenderObject type (e.g. `Gui` or `Debug`) without reaching, at some day in the future, the same problem or having to `hack` some behavior (by, e.g. adding a multiplier somewhere that depends on the number of light sources ) in dedicated applications... 
Even if the "new renderer" will integrate an "unlit" pass, acting only on materials tagged as unlit, this will not solve the problem in an efficient way for now. 
To do that, we first should redesign the `Renderer` interface, implemented by any concrete Renderer, to have a better classification of components to drawn and management of the changes in this classification, so that it will be efficient, pervasive and robust. 
This could be one reason to make the Renderer a System (as in Entity/Component/System pattern ^^) that manages its component using Observer/Observable pattern with RenderObjects promoted to the status of Components ... BTW, even if there is numerous other advantages to do so, this is a very huge redesign of Radium and should take time ...

**Comment by dlyr** [comment]
I'm not sure to get what is the correct way to have an object drawn without lighting i.e. how can I simply ignore lights, and hence light pass, on the "material"/shader side ?
My question is not related to any current implementation, but a more general design question.
If the answer is "this kind of object are UI" then it's okey for my case.

**Reply by MathiasPaulin** [reply]
Yes, we agree on the answer : this kind of objects are UI.
But this does not remove the need of a better `Ra::Engine::Rendering::RenderObjectType` management by the renderers.
